### PR TITLE
docs: Add docstring examples `mixture` estimators

### LINKED ISSUE #596: [DOCS] Example usage in docstring
As [discussed in #586](https://github.com/koaning/scikit-lego/pull/586#discussion_r1385475426), although most of the library features are documented in the user guide, the best way to showcase how to access and use each class/function would be a minimal example usage in the docstrings. And this is currently covered only for a subset of the library features.

Here the list of all the remaining classes that would benefit from it:

- [ ] `linear_model.LowessRegression`
- [ ] `linear_model.ProbWeightRegression`
- [ ] `linear_model.DeadZoneRegressor`
- [ ] `linear_model.DemographicParityClassifier`
- [ ] `linear_model.EqualOpportunityClassifier`
- [ ] `model_selection.TimeGapSplit`
- [ ] `model_selection.GroupTimeSeriesSplit`
- [ ] `model_selection.KlusterFoldValidation`
- [ ] `naive_bayes.GaussianMixtureNB`
- [ ] `naive_bayes.BayesianGaussianMixtureNB`
- [ ] `neighbors.BayesianKernelDensityClassifier`
- [ ] `decomposition.pca_reconstruction.PCAOutlierDetection`
- [ ] `decomposition.umap_reconstruction.UMAPOutlierDetection`
- [ ] `meta.confusion_balancer.ConfusionBalancer`
- [ ] `meta.estimator_transformer.EstimatorTransformer`
- [ ] `meta.grouped_predictor.GroupedPredictor`
- [ ] `meta.grouped_transformer.GroupedTransformer`
- [x] `meta.outlier_classifier.OutlierClassifier`
- [ ] `meta.regression_outlier_detector.RegressionOutlierDetector`
- [ ] `meta.subjective_classifier.SubjectiveClassifier`
- [ ] `meta.thresholder.Thresholder`
- [ ] `mixture.bayesian_gmm_classifier.BayesianGMMClassifier`
- [ ] `mixture.bayesian_gmm_detector.BayesianGMMOutlierDetector`
- [ ] `mixture.gmm_classifier.GMMClassifier`
- [ ] `mixture.gmm_detector.GMMOutlierDetector`
- [x] `preprocessing.dictmapper.DictMapper`
- [ ] `preprocessing.formulaictransformer.FormulaicTransformer`
- [ ] `preprocessing.identitytransformer.IdentityTransformer`
- [ ] `preprocessing.intervalencoder.IntervalEncoder`
- [x] `preprocessing.outlier_remover.OutlierRemover`
- [ ] `preprocessing.pandastransformers.PandasTypeSelector`
- [ ] `preprocessing.projections.InformationFilter`
- [ ] `preprocessing.repeatingbasis.RepeatingBasisFunction`

As an instance of such minimal example you can refer to [`QuantileRegression` docstring section](https://github.com/koaning/scikit-lego/blob/main/sklego/linear_model.py#L1140), which renders as in its [API section](https://koaning.github.io/scikit-lego/api/linear-model/#sklego.linear_model.QuantileRegression).

If possible try to add **one** unique example covering the relevant features and methods in the top level docstring of the class.

### PR DESCRIPTION
**Pull request by mkalimeri**
Related issue #596

Description: 

Added examples to models in 'mixture' folder

- `GMMClassifier`
- `BayesianGMMClassifier`
- `GMMOutlierDetector`
- `BayesianGMMOutlierDetector`

Type of change: Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: sklego/mixture/bayesian_gmm_classifier.py [hunk 1]
```diff
@@ -22,6 +22,48 @@ class BayesianGMMClassifier(BaseEstimator, ClassifierMixin):
         A dictionary of Bayesian Gaussian Mixture Models, one for each class.
     classes_ : np.ndarray of shape (n_classes,)
         The classes seen during `fit`.
+
+      Examples
+    --------
```
Comment (FBruzzesi) [inline sklego/mixture/bayesian_gmm_classifier.py:27]: Indentation should be:

```suggestion
    Examples
    --------
```

File: sklego/mixture/bayesian_gmm_classifier.py [hunk 2]
```diff
@@ -22,6 +22,48 @@ class BayesianGMMClassifier(BaseEstimator, ClassifierMixin):
...
+
+    ```python
+    import numpy as np
+    import matplotlib.pyplot as plt
```
Comment (FBruzzesi) [inline sklego/mixture/bayesian_gmm_classifier.py:31]: I think we can skip everything involving plotting, it won't be rendered anyway

File: sklego/mixture/bayesian_gmm_classifier.py [hunk 3]
```diff
@@ -22,6 +22,48 @@ class BayesianGMMClassifier(BaseEstimator, ClassifierMixin):
...
+    # Create and fit the BayesianGMMClassifier model
+    bgmm = BayesianGMMClassifier(n_components=2, random_state=1)
+    bgmm.fit(data, y)
+
+    # Classify the train dataset into two clusters (n_components=2)
+    labels = bgmm.predict(data)
+
+    # Classify a new point into one of two clusters
+    p = np.array([[1.5, 0.5]])
+    p_prob= bgmm.predict_proba(p) # predict the probabilities p belongs to each cluster
```
Comment (FBruzzesi) [inline sklego/mixture/bayesian_gmm_classifier.py:50]: Tiny format:

```suggestion
    p_prob = bgmm.predict_proba(p) # predict the probabilities p belongs to each cluster
```

File: sklego/mixture/bayesian_gmm_classifier.py [hunk 4]
```diff
@@ -22,6 +22,48 @@ class BayesianGMMClassifier(BaseEstimator, ClassifierMixin):
...
+    plt.scatter(group0[:,0], group0[:,1], c='y', label='group0')
+    plt.scatter(group1[:,0], group1[:,1], c='r', label='group1')
+    plt.scatter(p[:,0], p[:,1], c='black', marker='x', label='new datapoint')
+    plt.title('Distribution of dataset')
+    plt.legend()
+    plt.show()
+
+    ### Probability point p belongs to group1 is 0.38
+    ### Probability point p belongs to group2 is 0.62
+    ### It is more probable that point p belongs to group1
```
Comment (FBruzzesi) [inline sklego/mixture/bayesian_gmm_classifier.py:65]: I would add the print result under each print statement and remove everything involving plotting:

```suggestion
    print(f'Probability point p belongs to group1 is {p_prob[0,0]:.2f}')
    # Probability point p belongs to group1 is 0.38
    print(f'Probability point p belongs to group2 is {p_prob[0,1]:.2f}')
    # Probability point p belongs to group2 is 0.62

    print(f'It is more probable that point p belongs to group{np.argmax(p_prob)}')
    # It is more probable that point p belongs to group1
```

**Review by FBruzzesi — COMMENTED** [review]
Hey @mkalimeri , thanks for the PR and for your patience! Work has been a bit hectic recently.

This is very valuable and I would say that it is almost ready - I left comments on `bayesian_gmm_classifier.py`, but similar feedback is true for all examples:

- `Examples` indentation
- Avoid plotting functionalities in the docstrings
- Some minor code formatting
- Print results under each print statement

**Comment by mkalimeri** [comment]
Hello @FBruzzesi, thank you for the feedback! Working on issues is fun, I am happy to pick up more ;-) 

I committed the changes requested, but I see the tests are not passing. The errors appear to be related to other files, so I am not sure about how to proceed! Please let me know how things look. Thank you!

**Comment by FBruzzesi** [comment]
Hey @mkalimeri thanks for fixing the issues! I will review it soon-ish!

Don't worry about the test suite, we knew that scikit-learn 1.6 release would have broken our tests 🙃 we are working on a fix

**Review by FBruzzesi — APPROVED** [review]
Thanks @mkalimeri for the patience!
We fixed the issue with ci/cd and I pushed a fix for one example indentation! 

Ready to ship this 🚀

**Pull request merged**